### PR TITLE
Dataviews & Widgets guide

### PR DESCRIPTION
**Pull request by borja-munoz**
Initial version for the guide and examples

### CODE REVIEW

File: examples/guides/dataviews-widgets/step-3.html [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                    id="widgetCategory"
+                    heading="Countries by continent (on screen)"
+                    description="Number of countries by continent">
+                  </as-category-widget>
+                </section>
+                <section class="as-box">
+                    <as-histogram-widget
+                      id="widgetHistogram"
+                      heading="Countries by population"
+                      description="Distribution of countries by population">
```
Comment (manmorjim) [inline examples/guides/dataviews-widgets/step-3.html:54]: I would add the option `show-clear` on the widgets in order to allow users clear filters applied like the examples `examples/widgets`

Comment (manmorjim) [inline examples/guides/dataviews-widgets/step-3.html]: I would add the option `show-clear` on the widgets in order to allow users clear filters applied like the examples `examples/widgets`

**Review by manmorjim — APPROVED** [review]
LGTM! :ok_hand: 

Just two suggestions:

- Showing the clear button on widgets to allow users remove filters.
- What about adding an example using Dataviews like in `examples/dataview/viewport-features.html`? In the three examples the dataviews are used just as parameters for widgets which makes users ask theirself about the purpose of the dataviews.

File: docs/guides/04-dataviews-widgets.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+We have three different types of dataviews:
+
+* `carto.viz.dataview.Formula`
+* `carto.viz.dataview.Category`
+* `carto.viz.dataview.Histogram`
+
+When we create the dataview, we specify the data source, the feature property/field used for calculations and some additional options. These are the options that are available for all dataviews:
+
+* `mode`. When you are using a CARTO dataset as the data source, you can use the information from the vector tiles in the client to perform the dataview calculations or you can perform the calculations server-side. The tiles might not include all the existing features in the dataset because there are some aggregations that happen while creating the tiles to speed-up the visualization. This means that some operations (i.e. average) must not yield exact results. If exact results are needed, you must use server-side calculations. This option controls the calculation mode and the possible values are `DataViewCalculation.FAST` for using the tiles and `DataViewCalculation.PRECISE` for performing server-side calculations.
```
Comment (manmorjim) [inline docs/guides/04-dataviews-widgets.md]: About the modes, the users who use the `dist.min.js` should specify the values `'fast'` and `'precise'` similar to the `operation` parameter.

File: docs/guides/04-dataviews-widgets.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+The `carto.viz.dataview.Formula` and `carto.viz.dataview.Category` dataviews provide an option to specify the aggregation operation to execute on the features, through the option called `operation`. The possible values are:
+  * `'count'`
+  * `'avg'`
+  * `'min'`
+  * `'max'`
+  * `'sum'`
+  * `'percentile'` 
+
+In addition to these options, there are other options specifice to some dataviews. Please go to the reference to see the options available for each dataview.
```
Comment (manmorjim) [inline docs/guides/04-dataviews-widgets.md]: typo: "`specifice`"

**Review by manmorjim — APPROVED** [review]

File: docs/guides/04-dataviews-widgets.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+  <head>
+
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Getting Started Guide - Step 3</title>
```
Comment (VictorVelarde) [inline docs/guides/04-dataviews-widgets.md]: Title should be updated here

File: examples/guides/dataviews-widgets/step-3.html (hunk 1 above)
Comment (borja-munoz) [reply examples/guides/dataviews-widgets/step-3.html]: Added

File: docs/guides/04-dataviews-widgets.md (hunk 2 above)
Comment (borja-munoz) [reply docs/guides/04-dataviews-widgets.md]: Fixed

File: docs/guides/04-dataviews-widgets.md (hunk 3 above)
Comment (borja-munoz) [reply docs/guides/04-dataviews-widgets.md]: Fixed

File: docs/guides/04-dataviews-widgets.md (hunk 4 above)
Comment (borja-munoz) [reply docs/guides/04-dataviews-widgets.md]: Fixed

**Pull request merged**